### DISCUSSION TITLE
Toolchain files should modify `CMAKE_SYSTEM_PREFIX_PATH`, not `CMAKE_PREFIX_PATH`.

### DISCUSSION THREAD

**Discussion by jbadwaik** [microsoft/vcpkg#35642] [Ideas]
As seen [here](https://github.com/microsoft/vcpkg/blob/16ee2ecb31788c336ace8bb14c21801efb6836e4/scripts/buildsystems/vcpkg.cmake#L433) and [here](https://github.com/microsoft/vcpkg/blob/16ee2ecb31788c336ace8bb14c21801efb6836e4/scripts/buildsystems/vcpkg.cmake#L449), `vcpkg.cmake` modifies `CMAKE_PREFIX_PATH`. However, according to the [documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html), `CMAKE_PREFIX_PATH` belongs to the project and toolchains should instead set or modify `CMAKE_SYSTEM_PREFIX_PATH`.

**Comment by Neumann-A** [comment]
I think this boils done to personal preferences. I see vcpkg as part of the project and not as part of the system.

**Comment by jbadwaik** [comment]
I disagree. The general meaning of a project is the top-level `CMakeLists.txt` which has the `project` command or a file included from that top-level file. `vcpkg.cmake` is not used like that.

**Comment by dg0yt** [comment]
What we do want is a controlled and controllable search order/priority. 
But do we agree on this order?
1. user-provided locations (passed to cmake invocation)
2. project-provided locations (from `CMakeLists.txt`)
3. vpckg-provided locations (from `vcpkg.cmake`)
4. system-/toolchain-provided locations (from chainloaded toolchain file)

The question is how to establish the desired order reliably.
- For the first, there is just direct initalization of `CMAKE_...` user variables.
- The second can generally be safely handled by appending to the user variable, because it is applied after user input.  
  However, we have no control if it is before or after the `project` command which pulls in the toolchains.
- And that's why I think that `vcpkg.cmake` should prepend to `CMAKE_SYSTEM_...` variables, after loading the chainloaded toolchain file:
  This would ensure that vcpkg-provided locations take the right place.
  ATM `vcpkg.cmake` normally\* prepends to `CMAKE_...` user variables, so the user's explicit priorities are pushed down, and entries added by the project might be either before or after the vcpkg entries.
  \*) In the deprecated `VCPKG_PREFER_SYSTEM_LIBS` mode, `vcpkg.cmake` appends to the `CMAKE_...`user variables.
- OTOH people might be abusing the `CMAKE_` user variables to set search paths which actually belong to the toolchain/system/sysroot.

**Reply by jbadwaik** [reply]
I agree. You have laid it out nicely.
